shared/src/metabase/mbql/normalize.cljc

```diff
@@ -302,6 +302,9 @@
                      :source-query    normalize-source-query
                      :source-metadata {::sequence normalize-source-metadata}
                      :joins           {::sequence normalize-join}}
+   ;; we smuggle metadata for datasets and want to preserve their "database" form vs a normalized form so it matches
+   ;; the style in annotate.clj
+   :middleware      {:metadata/dataset-metadata identity}
```

Comment:
I put the existing dataset metadata in the middleware. But I might put it in `:info` if anyone prefers that. This is only set when you actually query a dataset on its own, not when it is a source of another query. This is the source of the db metadata and ensures it can persist and be merged into the query results metadata

Reply:
I have since moved this into info.

Reply:
I look at the difference like this: `info` is stuff that is _not_ passed in by the User running the query (and we wouldn't want it to be) -- stuff like permissions info (for sandboxing) or the Card/Dashboard ID we're running in the context of. If this was passed in via `:query` it could break stuff if we weren't careful to filter it out if it was user-supplied.

It's probably better if metadata from the application DB goes in `info` rather than `query`, since it could be a potential attack surface if users could pass it in as part of a request -- different metadata might affect how we do auto-bucketing or something like that -- and there's not a good reason to do so anyway.

So I think `info` makes sense here 👍